src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterModule.java

```diff
@@ -187,10 +189,31 @@ public EidasMatchingDatasetUnmarshaller getEidasMatchingDatasetUnmarshaller() {
     @Singleton
     public EidasUnsignedMatchingDatasetUnmarshaller getEidasUnsignedMatchingDatasetUnmarshaller(
             SecretKeyDecryptorFactory secretKeyDecryptorFactory,
-            MetadataResolverRepository eidasMetadataResolverRepository) {
+            MetadataResolverRepository eidasMetadataResolverRepository,
+            InstantValidator instantValidator,
+            SubjectValidator subjectValidator,
+            CountryConditionsValidator conditionsValidator,
+            @Named("AllAcceptableHubConnectorEntityIds") List<String> acceptableHubConnectorEntityIds
+            ) {
         StringToOpenSamlObjectTransformer<Response> stringtoOpenSamlObjectTransformer = new CoreTransformersFactory().getStringtoOpenSamlObjectTransformer(new ResponseSizeValidator());
         EidasValidatorFactory eidasValidatorFactory = new EidasValidatorFactory(eidasMetadataResolverRepository);
-        return new EidasUnsignedMatchingDatasetUnmarshaller(secretKeyDecryptorFactory, stringtoOpenSamlObjectTransformer, eidasValidatorFactory);
+
+        ExceptionThrowingValidator<Assertion> eidasSamlValidator = a -> {
+            try {
+                instantValidator.validate(a.getIssueInstant(), "Country Assertion IssueInstant");
+                // FIXME: the expectedInResponseTo param is never used in the subjectValidator
+                subjectValidator.validate(a.getSubject(), a.getID());
+                conditionsValidator.validate(a.getConditions(), acceptableHubConnectorEntityIds.toArray(new String[0]));
```

Comment:
What happens if `acceptableHubConnectorEntityIds`  has more than one entry? Does `toArray(new String[0])` just take the first one? Should we be looping through them all and validating to see if any are valid?

Reply:
I trust that this will translate all list elements into the correctly sized array, and this method has been used before.